src/widgets/ContentEditable.js

```diff
@@ -0,0 +1,314 @@
+import $ from 'jquery';
+import _ from 'lodash';
+import KeyCodes from 'keycodes-enum';
+import TimelineMax from 'TimelineMax';
+import AccessibilityModule from '@curriculumassociates/createjs-accessibility';
+
+const PAD = 2;
+const MODES = {
+  INSERT: 0,
+  OVERWRITE: 1,
+};
+
+export default class ContentEditable extends createjs.Container {
+  constructor(width, height, tabIndex) {
+    super();
+    _.bindAll(this, 'onFocus', 'onBlur', '_onCanvasKeyDown', '_onValueChanged', '_onSelectionChanged', '_onMouseDown', '_onMouseMove', '_onMouseUp');
+    console.log(tabIndex);
+    AccessibilityModule.register({
+      accessibleOptions: { tabIndex },
+      displayObject: this,
+      role: AccessibilityModule.ROLES.SPAN,
+    });
+
+    // contentEditable.accessible.tabIndex = this._nextTab++;
+    this.accessible.contenteditable = true;
+    this.addEventListener('valueChanged', this._onValueChanged);
+    this.addEventListener('selectionChanged', this._onSelectionChanged);
+    this.addEventListener('focus', this.onFocus);
+    this.addEventListener('blur', this.onBlur);
+    this.addEventListener('mousedown', this._onMouseDown);
+    this.addEventListener('pressmove', this._onMouseMove);
+    this.addEventListener('pressup', this._onMouseUp);
+
+    this.setBounds(0, 0, width, height);
+
+    const bg = new createjs.Shape();
+    bg.graphics.beginStroke('#000000').setStrokeStyle(1).beginFill('#ffffff').drawRect(0, 0, width, height);
+    this.addChild(bg);
+
+    this._selectionDisplay = new createjs.Shape();
+    this._selectionDisplay.visible = false;
+    this.addChild(this._selectionDisplay);
+
+    const fontSize = height - PAD * 2;
+    this._text = new createjs.Text('', `${fontSize}px Arial`);
+    this._text.x = PAD;
+    this._text.y = PAD;
+    this.addChild(this._text);
+    this.accessible.value = this._text;
+    this._cursor = new createjs.Shape();
+    this._cursor.graphics.beginStroke('#000000').setStrokeStyle(1).moveTo(0, PAD).lineTo(0, fontSize);
+    this._cursor.x = PAD;
+    this._cursor.visible = false;
+    this.addChild(this._cursor);
+
+    this._cursorIndex = 0;
+    this._selection = { start: -1, end: -1 };
+    this._height = height;
+
+    this.addEventListener('click', this.onFocus);
+    // todo: handle mouse based blur causing event
```

Comment:
Looks like this is copied from the SingleLineTextInput or MultiLineTextInput classes and that is a part that is planned to be refactored by moving focus onto the semantic text input element instead of using a custom key event listener.  To handle the canvas side of this test case, I would recommend inheriting from one of those classes and overriding only what is needed.

Reply:
removed this global attribute

Reply:
removed